CodingTracker.AlvinGraham/CodingTracker.AlvinGraham/DataAccess.cs

```diff
@@ -0,0 +1,167 @@
+﻿using Dapper;
+using Microsoft.Data.Sqlite;
+using Microsoft.Extensions.Configuration;
+
+namespace CodingTracker;
+
+internal class DataAccess
+{
+	IConfiguration configuration = new ConfigurationBuilder()
+	.AddJsonFile("appsettings.json")
+	.Build();
+
+	private string ConnectionString;
+	public DataAccess()
+	{
+		ConnectionString = configuration.GetSection("ConnectionStrings")["DefaultConnection"]!;
+	}
+
+	internal void CreateDatabase()
+	{
+		using (var connection = new SqliteConnection(ConnectionString))
+		{
+			connection.Open();
+
+			string createTableQuery = @"
+			CREATE TABLE IF NOT EXISTS records (
+				Id INTEGER PRIMARY KEY AUTOINCREMENT,
+				DateStart TEXT NOT NULL,
+				DateEnd TEXT NOT NULL
+			)";
+
+			connection.Execute(createTableQuery);
+
+			createTableQuery = @"
+			CREATE TABLE IF NOT EXISTS goals (
+				Id INTEGER PRIMARY KEY AUTOINCREMENT,
+				TotalMinutes INTEGER NOT NULL,
+				DateEnd TEXT NOT NULL
+			)";
+
+			connection.Execute(createTableQuery);
+		}
+	}
+
+	internal void InsertRecord(CodingRecord record)
+	{
+		using (var connection = new SqliteConnection(ConnectionString))
+		{
+			connection.Open();
+
+			string insertQuery = @"
+			INSERT INTO records (DateStart, DateEnd)
+			VALUES (@DateStart, @DateEnd)";
+
+			connection.Execute(insertQuery, new { record.DateStart, record.DateEnd });
+		}
+	}
+
+	internal IEnumerable<CodingRecord> GetAllRecords()
+	{
+		using (var connection = new SqliteConnection(ConnectionString))
+		{
+			connection.Open();
+
+			string selectQuery = "SELECT * FROM records";
+
+			var records = connection.Query<CodingRecord>(selectQuery);
+
+			foreach (var record in records)
+			{
+				record.Duration = record.DateEnd - record.DateStart;
+			}
+
+			return records;
+		}
+	}
+
+	internal List<CodingRecord> GetRecordList()
```

Comment:
No need to create a function for List of records
`var records = connection.Query<CodingRecord>(selectQuery);`
You can get list as
`records.ToList()`

Reply:
I think I see what you are saying here.  In essence I had two methods that did the exact same thing but one returned a List and the other returned an IEnumerable.  I kept the IEnumerable one and just used .ToList() method where necessary.